packages/next-swc/crates/core/src/next_dynamic.rs

```diff
@@ -236,16 +236,48 @@ impl Fold for NextDynamicPatcher {
                             value: generated,
                         })))];
 
+                    let mut has_ssr_false = false;
+
                     if expr.args.len() == 2 {
                         if let Expr::Object(ObjectLit {
                             props: options_props,
                             ..
                         }) = &*expr.args[1].expr
                         {
+                            for prop in options_props.iter() {
+                                if let Some(KeyValueProp { key, value }) =
+                                    prop.clone().prop().and_then(|p| p.key_value())
+                                {
+                                    if key.ident().and_then(|i| (Some(i.sym.eq("ssr".into()))))
+                                        == Some(true)
+                                    {
+                                        if let Some(Bool { value, span: _ }) =
```

Comment:
Instead of checking if `value == false`, you can do `if let Some(Bool { value: false, span: _ }) =`.